### PR TITLE
 uidsl refactor

### PR DESCRIPTION
**Pull request by xurui1995**

### CODE REVIEW

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedAnnotationTab.kt [hunk 1]
```diff
@@ -15,90 +15,93 @@ import javax.swing.ScrollPaneConstants
...
+                    })
+                    jRadioButton("Others by customize", ConfigManager.targetJsonConverterLib == TargetJsonConverter.Custom, {
                         ConfigManager.targetJsonConverterLib = TargetJsonConverter.Custom
-                    }().addToGroup()
-
-                    onRadioButtonSelectListener = {
-                        callBack(it == customizeAnnotationRadioButton)
+                    }) {
+                        addChangeListener {
+                            callBack(isSelected)
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedAnnotationTab.kt:99]: It seems That the logic is not the same with before

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt [hunk 2]
```diff
@@ -11,51 +11,37 @@ import javax.swing.JPanel
...
-                    radioButton("Auto Determine Nullable Or Not From JSON Value", ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.AutoDeterMineNullableOrNot) {
-                        ConfigManager.propertyTypeStrategy = PropertyTypeStrategy.AutoDeterMineNullableOrNot
-                    }().addToGroup()
+                jLine()
+                jLabel("Type")
+                jButtonGroup {
+                    jRadioButton("Non-Nullable", ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.NotNullable,
+                            { ConfigManager.propertyTypeStrategy = PropertyTypeStrategy.NotNullable })
+                    jRadioButton("Nullable", ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.Nullable,
+                            { ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.Nullable })
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt:30]: Here `==` should be `=`, All Right?

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt [hunk 3]
```diff
@@ -11,51 +11,37 @@ import javax.swing.JPanel
...
+                    jRadioButton("Auto Determine Nullable Or Not From JSON Value", ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.AutoDeterMineNullableOrNot,
+                            { ConfigManager.propertyTypeStrategy == PropertyTypeStrategy.AutoDeterMineNullableOrNot })
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt:32]: The same problem before

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt [hunk 4]
```diff
@@ -11,51 +11,37 @@ import javax.swing.JPanel
...
-                        ConfigManager.defaultValueStrategy = DefaultValueStrategy.AvoidNull
-                    }().addToGroup()
-                    radioButton("Init With Default Value Null When Property Is Nullable", ConfigManager.defaultValueStrategy == DefaultValueStrategy.AllowNull) {
-                        ConfigManager.defaultValueStrategy = DefaultValueStrategy.AllowNull
-                    }().addToGroup()
+                jLine()
+                jLabel("Default Value Strategy")
+                jButtonGroup {
+                    jRadioButton("Don't Init With Default Value", ConfigManager.defaultValueStrategy == DefaultValueStrategy.None,
+                            { ConfigManager.defaultValueStrategy == DefaultValueStrategy.None })
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt:38]: The same

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt [hunk 5]
```diff
@@ -11,51 +11,37 @@ import javax.swing.JPanel
...
+                    jRadioButton("Init With Non-Null Default Value (Avoid Null)", ConfigManager.defaultValueStrategy == DefaultValueStrategy.AvoidNull,
+                            { ConfigManager.defaultValueStrategy == DefaultValueStrategy.AvoidNull })
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt:40]: The same

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt [hunk 6]
```diff
@@ -11,51 +11,37 @@ import javax.swing.JPanel
...
+                    jRadioButton("Init With Default Value Null When Property Is Nullable", ConfigManager.defaultValueStrategy == DefaultValueStrategy.AllowNull,
+                            { ConfigManager.defaultValueStrategy == DefaultValueStrategy.AllowNull })
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedPropertyTab.kt:42]: The same

File: src/main/kotlin/wu/seal/jsontokotlin/ui/JsonInputDialog.kt [hunk 7]
```diff
@@ -88,31 +89,55 @@ class JsonInputDialog(classsName: String, private val project: Project) : Messag
...
+                            sendActionInfo(prettyGson.toJson(ClickProjectURLAction()))
+                        }
+                    }
+                }
+            }
         }
     }
 
+
     private fun createAdvancedPanel(): JPanel {
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/ui/JsonInputDialog.kt:132]: Should this method been removed?

File: src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedAnnotationTab.kt (hunk 1 above)
Comment (xurui1995) [reply src/main/kotlin/wu/seal/jsontokotlin/ui/AdvancedAnnotationTab.kt:99]: I think the logic is the same as before. just simpler than before.

**Pull request merged**